src/detext/train/train.py

```diff
@@ -298,6 +303,18 @@ def model_fn(features, labels, mode, params):
 
     group_size_field = features['group_size'] if mode != tf.estimator.ModeKeys.PREDICT else None
 
+    # For multitask training
+    task_id_field = features.get('task_id', None)  # shape=[batch_size,]
+
+    # Update the weight with each task's weight such that weight per document = weight * task_weight
+    if params.task_ids is not None:
+        task_ids = params.task_ids  # e.g. [0, 1, 2]
+        task_weights = params.task_weights  # e.g. [0.1, 0.3, 0.6]
+        # Expand task_id_field with shape [batch_size, num_tasks]
+        expanded_task_id_field = tf.transpose(tf.broadcast_to(task_id_field, [len(task_ids), tf.shape(task_id_field)[0]]))
+        task_mask = tf.cast(tf.equal(expanded_task_id_field, task_ids), dtype=tf.float32)
```

Comment:
Just make sure that results are correct here with the implicit broadcasting of task_ids and task_weights

Reply:
Thanks. I tested locally this part with sample data. Did not add the unit test as it's more about tf.broadcast_to practice.